.github/workflows/build-test-distribute.yaml

```diff
@@ -0,0 +1,240 @@
+        if: ${{ ! contains(github.event.pull_request.labels.*.name, 'ci/skip-container-structure-test') }}
+        with:
+          name: build-output
+      - name: Unpack artifact
+        run: |
+          tar -xzf build.tar.gz
+      - name: Prepare for container structure test
+        if: ${{ ! contains(github.event.pull_request.labels.*.name, 'ci/skip-container-structure-test') }}
+        run: |
+          sudo apt-get update; sudo apt-get install -y qemu-user-static binfmt-support
+      - name: Run container structure test
+        if: ${{ ! contains(github.event.pull_request.labels.*.name, 'ci/skip-container-structure-test') }}
+        run: |
+          make test/container-structure
+  distributions:
+    needs: test
+    # this jobs will still run even if "test" is skipped
+    if: ${{ always() }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check if force push"
+        if: ${{ contains(github.event.pull_request.labels.*.name, 'ci/force-publish') }}
+        run: 'echo "ALLOW_PUSH=true" >> $GITHUB_ENV'
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: go.mod
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.kuma-dev
+          key: ${{ runner.os }}-devtools-${{ hashFiles('mk/dependencies/deps.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-devtools
+      - uses: actions/download-artifact@v2
+        with:
+          name: build-output
+      - name: Unpack artifact
+        run: |
+          tar -xzf build.tar.gz
+      - name: Inspect created tars
+        run: |
+          for i in build/distributions/out/*.tar.gz; do echo $i; tar -tvf $i; done
+      - name: Publish distributions to Pulp
+        run: |
+          make publish/pulp
+      - name: Load images
+        run: |
+          make docker/load
+      - name: Publish images
+        run: |-
+          make docker/login
+          # ensure we always logout
+          function on_exit() {
+            make docker/logout
+          }
+          trap on_exit EXIT
+          make docker/push
+          make docker/manifest
+  e2e_legacy:
+    name: "legacy-k8s:${{ matrix.arch }}-${{ matrix.k8sVersion }}"
+    needs: build
+    uses: kumahq/kuma/.github/workflows/e2e.yaml@master
+    strategy:
+      matrix:
+        k8sVersion:
+          - kind
+          - kindIpv6
+          - v1.23.17-k3s1
+          - v1.28.1-k3s1
+        arch:
+          - amd64
+          - arm64
+    with:
+      k8sVersion: ${{ matrix.k8sVersion }}
+      arch: ${{ matrix.arch }}
+      parallelism: 3
+      target: ""
+  e2e_main:
+    name: "${{ matrix.target }}:${{ matrix.arch }}-${{ matrix.k8sVersion }}"
+    needs: build
+    uses: kumahq/kuma/.github/workflows/e2e.yaml@master
+    strategy:
+      matrix:
+        k8sVersion:
+          - kind
+          - kindIpv6
+          - v1.23.17-k3s1
```

Comment:
We can't reference a variable here as GitHub Actions only supporting reference variables for matrix (by `inputs`) in a reusable workflow file.